File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/epg/retrofit/EpgPresetResponseBodyConverter.java
package org.dvbviewer.controller.data.epg.retrofit;


import org.dvbviewer.controller.entities.EpgEntry;
import org.dvbviewer.controller.io.data.EpgEntryHandler;

import java.util.List;

import okhttp3.ResponseBody;
import retrofit2.Converter;

public class EpgPresetResponseBodyConverter implements Converter<ResponseBody, List<EpgEntry>> {

    private final EpgEntryHandler prefsHandler = new EpgEntryHandler();

    @Override public List<EpgEntry> convert(ResponseBody value) {
        try {
            return prefsHandler.parse(value.byteStream());
        } catch (RuntimeException e) {
            throw e;
        } catch (Exception e) {
            throw new RuntimeException(e);
        } finally {
            value.close();
        }
    }

}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/epg/retrofit/EpgPresetResponseBodyConverter.kt
package org.dvbviewer.controller.data.epg.retrofit


import okhttp3.ResponseBody
import org.dvbviewer.controller.data.api.handler.EpgEntryHandler
import org.dvbviewer.controller.data.entities.EpgEntry
import retrofit2.Converter

class EpgPresetResponseBodyConverter : Converter<ResponseBody, List<EpgEntry>> {

    private val prefsHandler = EpgEntryHandler()

    override fun convert(value: ResponseBody): List<EpgEntry>? {
        try {
            return prefsHandler.parse(value.byteStream())
        } catch (e: RuntimeException) {
            throw e
        } catch (e: Exception) {
            throw RuntimeException(e)
        } finally {
            value.close()
        }
    }

}